z3/pput.py

```diff
@@ -278,8 +278,8 @@ def parse_args():
         description='Read data from stdin and upload it to s3',
         epilog=('All optional args have a configurable default. '
                 'Order of precedence is command line args then '
-                'environment variables then user config ~/.z3.cfg'
-                ' then default config.'),
+                'user config ~/.z3.cfg then environment variables '
```

Comment:
By reading the code it looks to me the env variables have precedence even over `~/.z3.cfg`, which IMHO they should.